x-pack/plugins/inference/server/chat_complete/adapters/bedrock/bedrock_claude_adapter.ts

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { from, switchMap, tap, filter, map } from 'rxjs';
+import { Readable } from 'stream';
+import type { InvokeAIActionParams } from '@kbn/stack-connectors-plugin/common/bedrock/types';
+import { parseSerdeChunkMessage } from './serde_utils';
+import { Message, MessageRole } from '../../../../common/chat_complete';
+import { createInferenceInternalError } from '../../../../common/errors';
+import { ToolChoiceType, type ToolOptions } from '../../../../common/chat_complete/tools';
+import { InferenceConnectorAdapter } from '../../types';
+import type { BedRockMessage, BedrockToolChoice } from './types';
+import {
+  BedrockChunkMember,
+  serdeEventstreamIntoObservable,
+} from './serde_eventstream_into_observable';
+import { processCompletionChunks } from './process_completion_chunks';
+
+export const bedrockClaudeAdapter: InferenceConnectorAdapter = {
+  chatComplete: ({ executor, system, messages, toolChoice, tools }) => {
+    const connectorInvokeRequest: InvokeAIActionParams = {
+      system,
+      messages: messagesToBedrock(messages),
+      tools: toolsToBedrock(tools),
+      toolChoice: toolChoiceToBedrock(toolChoice),
+      temperature: 0,
+      stopSequences: ['\n\nHuman:'],
+    };
+
+    return from(
+      executor.invoke({
+        subAction: 'invokeStream',
+        subActionParams: connectorInvokeRequest,
+      })
+    ).pipe(
+      switchMap((response) => {
+        const readable = response.data as Readable;
+        return serdeEventstreamIntoObservable(readable);
+      }),
+      tap((eventData) => {
+        if ('modelStreamErrorException' in eventData) {
+          throw createInferenceInternalError(eventData.modelStreamErrorException.originalMessage);
+        }
+      }),
+      filter((value): value is BedrockChunkMember => {
+        return 'chunk' in value && value.chunk?.headers?.[':event-type']?.value === 'chunk';
+      }),
+      map((message) => {
+        return parseSerdeChunkMessage(message.chunk);
+      }),
+      processCompletionChunks()
+    );
+  },
+};
+
+const toolChoiceToBedrock = (
+  toolChoice: ToolOptions['toolChoice']
+): BedrockToolChoice | undefined => {
+  if (toolChoice === ToolChoiceType.required) {
+    return {
+      type: 'any',
+    };
+  } else if (toolChoice === ToolChoiceType.auto) {
+    return {
+      type: 'auto',
+    };
+  } else if (typeof toolChoice === 'object') {
+    return {
+      type: 'tool',
+      name: toolChoice.function,
+    };
+  }
+  // ToolChoiceType.none is not supported by claude.
+  return undefined;
```

Comment:
in this case, should we append something to the system message?

Reply:
Yeah, I had it on my TODO and somehow forgot about it, I will add it

Reply:
https://github.com/elastic/kibana/pull/191434/commits/33f2a2b69455f8873fdf1f9c3bc12726f9103ec5